GliaWidgetsTests/Sources/EntryWidget/EntryWidgetTests.swift

```diff
@@ -56,25 +61,97 @@ class EntryWidgetTests: XCTestCase {
             return UUID().uuidString
         }
         let queuesMonitor = QueuesMonitor(environment: queueMonitorEnvironment)
-        var engagementLauncher = EngagementLauncher { _, _ in }
-        var isAuthenticated: () -> Bool = { true }
+        let engagementLauncher = EngagementLauncher { _, _ in }
+        let isAuthenticated: () -> Bool = { true }
 
         let entryWidget = EntryWidget(
             queueIds: [mockQueueId],
             environment: .init(
+                observeSecureUnreadMessageCount: { _ in UUID().uuidString },
+                unsubscribeFromUpdates: { _, _ in },
                 queuesMonitor: queuesMonitor,
                 engagementLauncher: engagementLauncher,
                 theme: Theme(),
+                log: .mock,
                 isAuthenticated: isAuthenticated
             )
         )
 
         entryWidget.show(in: .init())
 
         if case let .mediaTypes(mediaTypes) = entryWidget.viewState {
-            XCTAssertTrue(mediaTypes.contains(.secureMessaging))
+            XCTAssertTrue(mediaTypes.contains(.make(type: .secureMessaging)))
+        } else {
+            XCTFail("Unexpected view state")
+        }
+    }
+    
+    func test__secureMessagingIsShownWhenUserIsAuthenticatedAndHasUnreadMessages() {
+        let mockQueueId = "mockQueueId"
+        let mockQueue = Queue.mock(id: mockQueueId, media: [.messaging, .audio])
+        var queueMonitorEnvironment: QueuesMonitor.Environment = .mock
+        queueMonitorEnvironment.listQueues = { completion in
+            completion([mockQueue], nil)
+        }
+        queueMonitorEnvironment.subscribeForQueuesUpdates = { _, completion in
+            completion(.success(mockQueue))
+            return UUID().uuidString
+        }
+        let queuesMonitor = QueuesMonitor(environment: queueMonitorEnvironment)
+
+        let engagementLauncher = EngagementLauncher { _, _ in }
+        let isAuthenticated: () -> Bool = { true }
+        
+        let observeMessageCount: (_ completion: @escaping (Result<Int?, Error>) -> Void) -> String? = { completion in
+            completion(.success(5))
+            return UUID().uuidString
+        }
+
+        let entryWidget = EntryWidget(
+            queueIds: [mockQueueId],
+            environment: .init(
+                observeSecureUnreadMessageCount: observeMessageCount,
+                unsubscribeFromUpdates: { _, _ in },
+                queuesMonitor: queuesMonitor,
+                engagementLauncher: engagementLauncher,
+                theme: Theme(),
+                log: .mock,
+                isAuthenticated: isAuthenticated
+            )
+        )
+        
+        entryWidget.show(in: .init())
+
+        if case let .mediaTypes(mediaTypes) = entryWidget.viewState {
+            XCTAssertTrue(mediaTypes.contains(.make(type: .secureMessaging, badgeCount: 5)))
         } else {
             XCTFail("Unexpected view state")
```

Comment:
Can unexpected view state be pointed to in failure message?
```suggestion
            XCTFail("Unexpected view state: \(entryWidget.viewState)")
```